Description: Use pkg-config to get systemd libs to allow build on non-linux systems
Author: Roger Light <roger@atchoo.org>
Forwarded: yes
--- a/config.mk
+++ b/config.mk
@@ -242,8 +242,10 @@
 endif
 
 ifeq ($(WITH_SYSTEMD),yes)
-	BROKER_CPPFLAGS:=$(BROKER_CPPFLAGS) -DWITH_SYSTEMD
-	BROKER_LDADD:=$(BROKER_LDADD) -lsystemd
+	ifneq ($(shell pkg-config --libs libsystemd),)
+		BROKER_CPPFLAGS:=$(BROKER_CPPFLAGS) -DWITH_SYSTEMD $(shell pkg-config --cflags libsystemd)
+		BROKER_LDADD:=$(BROKER_LDADD) $(shell pkg-config --libs libsystemd)
+	endif
 endif
 
 ifeq ($(WITH_SRV),yes)
